pytest_django/plugin.py

```diff
@@ -399,12 +399,54 @@ def teardown():
         request.addfinalizer(teardown)
 
 
-@pytest.fixture(autouse=True, scope='function')
-def _django_clear_outbox(django_test_environment):
-    """Clear the django outbox, internal to pytest-django."""
-    if django_settings_is_configured():
-        from django.core import mail
-        del mail.outbox[:]
+class _DirectMailboxAccessProtector(list):
+
+    def _raise_assertion(*args, **kwargs):
+        raise AssertionError('To access mail.outbox, use the mailoutbox fixture.')
+
+    __len__ = __getitem__ = __nonzero__ = __bool__ = _raise_assertion
+
+
+# def mailoutbox_setup_and_teardown(outbox_type):
+#
+
+
+@pytest.yield_fixture(autouse=True)
+def _error_on_direct_mail_outbox_access():
+    if not django_settings_is_configured():
+        return
+
+    from django.core import mail
+
+    _old_mailbox = getattr(mail, 'outbox', None)
+    outbox = _DirectMailboxAccessProtector()
+    setattr(mail, 'outbox', outbox)
+
+    yield outbox
+
+    if _old_mailbox is not None:
+        setattr(mail, 'outbox', _old_mailbox)
+    else:
+        delattr(mail, 'outbox')
```

Comment:
mail.outbox will never be None, mail.outbox is only setup when locmem.EmailBackend is used, and setup either in constructor of locmem.EmailBackend or in django.test.util.setup_test_environment